### PR TITLE
Inline volumeMenuButton option

### PR DESCRIPTION
**Pull request by mmcc**
This adds a new `inline` option to `MenuButton` that allows you to use MenuButton as a drawer of sorts. Old functionality is unchanged, although the styles now reside under `.vjs-menu-button-popup`.

To use, simply pass `inline: true` in the `volumeMenuButton` options. This will also default to a horizontal slider for obvious reasons.

```javascript
videojs('vid1', { "playbackRates": [1, 2, 3], "controlBar": { "volumeMenuButton": { "inline": true }}});
```

There's an unrelated issue in IE8 where bubbling seems to be messed up. Everything works as far as this PR is concerned, but the bubbling issue is causing any click on the menu to count for a click on the mute button. We need to sort that out still, but I don't think this PR is the place for it.

preview

### CODE REVIEW

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 1b28a9329a1cac2915aa84c7d7d3e6e1fd736045
Build details: https://travis-ci.org/pam/video.js/builds/71510224

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by heff** [comment]
Very cool! 

How do I test it? I tried:
```
"controlBar": { "volumeMenuButton": { "inline": true } }
```
and got

screen shot 2015-07-18 at 11 20 26 am

It'd be nice if we could just do `volumeMenuButton": { "inline": true }` at the top level of the player options. Now that we're passing playerOptions to children we could probably do that easily. Don't try to add that to this PR though.

**Comment by mmcc** [comment]
@heff you also need to specify `vertical: false`. 

```json
"controlBar": { "volumeMenuButton": {"vertical": "false, "inline": true } }
```

**Comment by gkatsev** [comment]
Can `inline: true` imply `vertical`? I don't think there would be any case where you would want it to be inline and also vertical. And if you did, you could always set `vertical` to `true` to force it.

**Comment by mmcc** [comment]
`inline: true` can imply vertical for the `volumeMenuButton` specifically, but keep in mind that the option is intended to be generic to any instance of MenuButton.

**Comment by gkatsev** [comment]
That's true. But the `volumeMenuButton` can default to `vertical: false` when `inline: true`. I'm not certain that we do want this, but it does seem to simplify the simple base case.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 4b6bcdf1935f071731d8133bf40c24d30c4cf9cc
Build details: https://travis-ci.org/pam/video.js/builds/71802376

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mmcc** [comment]
@gkatsev @heff I went ahead and made the switch to set `vertical: false` if `inline: true` is specified on an instance of VolumeMenuButton. Tests are currently failing because of the node-sass issue on Travis.

File: src/js/control-bar/volume-menu-button.js [hunk 1]
```diff
@@ -19,6 +19,12 @@ import VolumeBar from './volume-control/volume-bar.js';
 class VolumeMenuButton extends MenuButton {
 
   constructor(player, options={}){
+    // If an inline volumeMenuButton is used, we should default to using a horizontal
+    // slider for obvious reasons.
+    if (options.inline) {
```
Comment (heff) [inline src/js/control-bar/volume-menu-button.js]: Might as well move this inside the `options.vertical === undefined` check, so it's overridable.

File: src/css/components/menu/_menu-inline.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
+.video-js .vjs-menu-button-inline {
```
Comment (heff) [inline src/css/components/menu/_menu-inline.scss]: I know it's a lot of CSS, but seems better to me to keep all the menu styles in one file. Just my opinion.

**Comment by heff** [comment]
A few more comments, not blockers. LGTM

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 4e68e6fe64d1b313053ce3f13458f8d1c7bccbf0
Build details: https://travis-ci.org/pam/video.js/builds/71813262

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: ad810aeddd9f3729007be3c6932a0f2086b1e29f
Build details: https://travis-ci.org/pam/video.js/builds/72034158

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 34a38c3a6e8f6d9350e56e6ce433d2b5ea7517ed
Build details: https://travis-ci.org/pam/video.js/builds/72039643

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 70e0c4877ea40cb9d824a167d219a7bee0d6a681
Build details: https://travis-ci.org/pam/video.js/builds/72042593

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mmcc** [comment]
Bumping this thread, I think this is good to pull in. @dmlap @gkatsev - thoughts?

**Comment by mmcc** [comment]
Also to note, I've personally tested this in:

OSX: Chrome, Firefox, Safari
Windows: Chrome, Firefox, IE8, IE11

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: a2c56a71107279168465facc27ec0654d3c43adb
Build details: https://travis-ci.org/pam/video.js/builds/72522493

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 7a842e224a3851d761c0cc73329da036357f6872
Build details: https://travis-ci.org/pam/video.js/builds/72525471

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by heff** [comment]
Nice, you ere able to fix IE8? LGTM

**Comment by mmcc** [comment]
@heff Everything related to this PR is working in IE8, but we still need to loop back on the bubbling issue.

**Comment by heff** [comment]
@mmcc can you create a new issue for the bubbling?

-- 
Sent from mobile

**Pull request closed without merging**